runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/resources/PipelineResourcesTest.java

```diff
@@ -28,51 +31,30 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
-import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
 import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.ExpectedException;
 import org.junit.rules.TemporaryFolder;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
-import org.mockito.Mockito;
 
 /** Tests for PipelineResources. */
 @RunWith(JUnit4.class)
 public class PipelineResourcesTest {
 
   @Rule public transient TemporaryFolder tmpFolder = new TemporaryFolder();
-  @Rule public transient ExpectedException thrown = ExpectedException.none();
 
   @Test
-  public void detectClassPathResourceWithFileResources() throws Exception {
+  public void testDetectsResourcesToStage() throws IOException {
     File file = tmpFolder.newFile("file");
-    File file2 = tmpFolder.newFile("file2");
-    URLClassLoader classLoader =
-        new URLClassLoader(new URL[] {file.toURI().toURL(), file2.toURI().toURL()});
+    URLClassLoader classLoader = new URLClassLoader(new URL[] {file.toURI().toURL()});
+    PipelineResourcesOptions options =
+        PipelineOptionsFactory.create().as(PipelineResourcesOptions.class);
 
-    assertEquals(
-        ImmutableList.of(file.getAbsolutePath(), file2.getAbsolutePath()),
-        PipelineResources.detectClassPathResourcesToStage(classLoader));
-  }
-
-  @Test
-  public void detectClassPathResourcesWithUnsupportedClassLoader() {
-    ClassLoader mockClassLoader = Mockito.mock(ClassLoader.class);
-    thrown.expect(IllegalArgumentException.class);
-    thrown.expectMessage("Unable to use ClassLoader to detect classpath elements.");
+    List<String> detectedResources =
+        PipelineResources.detectClassPathResourcesToStage(classLoader, options);
 
-    PipelineResources.detectClassPathResourcesToStage(mockClassLoader);
-  }
-
-  @Test
-  public void detectClassPathResourceWithNonFileResources() throws Exception {
```

Comment:
What about this case? How is it handled?

Reply:
The resource that is not a jar/zip file or directory will be ignored - this is the behavior of classgraph library. I moved the test to the `ClasspathScanningResourcesDetectorTest.java` class and changed it accordingly (but I forgot to change the name and I will do this in a followup commit 😶  ...). 

In case we still want to throw exceptions we'd need to detect this case on our own rather than relying on classpath behaviour. Is this needed?